### PR TITLE
Provider: Packet - add ip and volume

### PR DESCRIPTION
**Pull request by ayudemura**
We have updated our [packngo](https://github.com/packethost/packngo), and plugged in ip service and volume service.

### CODE REVIEW

File: builtin/providers/packet/provider.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ func Provider() terraform.ResourceProvider {
 			"packet_device":  resourcePacketDevice(),
 			"packet_ssh_key": resourcePacketSSHKey(),
 			"packet_project": resourcePacketProject(),
+			"packet_volume":  resourcePacketVolume(),
```
Comment (stack72) [inline builtin/providers/packet/provider.go]: Should this also include `packed_ip`?

Comment (ayudemura) [inline builtin/providers/packet/provider.go]: indeed, thanks 😸

**Comment by stack72** [comment]
Hi @ayudemura 

thanks for the PR here. Unfortunately, when I run some of the tests, I get failures:

```
=== RUN   TestAccPacketIPAddress_Basic
--- FAIL: TestAccPacketIPAddress_Basic (0.00s)
	testing.go:264: Step 0 error: Configuration is invalid.

		Warnings: []string(nil)

		Errors: []string{"packet_ip_address.foobar: \"address\": this field cannot be set"}
```

```
=== RUN   TestAccPacketIPReservation_Basic
--- FAIL: TestAccPacketIPReservation_Basic (0.00s)
	testing.go:264: Step 0 error: Configuration is invalid.

		Warnings: []string(nil)

		Errors: []string{"packet_ip_reservation.foobar: \"address\": this field cannot be set"}
```

```
=== RUN   TestAccPacketVolume_Basic
--- FAIL: TestAccPacketVolume_Basic (0.00s)
	testing.go:264: Step 0 error: Configuration is invalid.

		Warnings: []string(nil)

		Errors: []string{"packet_volume.foobar: \"name\": this field cannot be set", "packet_volume.foobar: \"facility\": required field is not set", "packet_volume.foobar: \"plan\": required field is not set", "packet_volume.foobar: \"billing_cycle\": required field is not set", "packet_volume.foobar: \"project_id\": required field is not set"}
```

Please can you have a look at these?

thanks

Paul

**Comment by ayudemura** [comment]
@stack72 
Thank you for your update.  When I run tests locally, I got no error...  I used `make test TEST=./builtin/providers/packet` command.  Can you give me the command you used when you see error?
Thanks!

**Comment by stack72** [comment]
Hi @ayudemura 

The `make test` doesn't run the acceptance tests. This command will run the entire suite:

```
make testacc TEST=./builtin/providers/packet TESTARGS='-run=TestAccPacket'
```

**Comment by stack72** [comment]
Hi @ayudemura 

Do you want me to retest this? :) I see you pushed a commit to fix the tests

Thanks

Paul

**Comment by ayudemura** [comment]
@stack72 
Yes, that would be great!  It "should" pass this time...
Thank you 😸

**Comment by stack72** [comment]
Hi @ayudemura 

Afraid not :( Still not working

```
% make testacc TEST=./builtin/providers/packet TESTARGS='-run=TestAccPacket'
==> Checking that code complies with gofmt requirements...
/Users/stacko/Code/go/bin/stringer
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/07/26 22:03:26 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/packet -v -run=TestAccPacket -timeout 120m
=== RUN   TestAccPacketIPAddress_Basic
--- FAIL: TestAccPacketIPAddress_Basic (0.62s)
	testing.go:264: Step 0 error: Error applying: 1 error(s) occurred:

		* packet_ip_address.foobar: Not found
=== RUN   TestAccPacketIPReservation_Basic
panic: interface conversion: interface is nil, not string

goroutine 136 [running]:
panic(0x737b20, 0xc820388000)
	/opt/boxen/homebrew/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
github.com/hashicorp/terraform/builtin/providers/packet.resourcePacketIPReservationCreate(0xc8202758c0, 0x7353c0, 0xc8201a0360, 0x0, 0x0)
	/Users/stacko/Code/go/src/github.com/hashicorp/terraform/builtin/providers/packet/resource_packet_ip_reservation.go:72 +0xd2
github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc82001af00, 0xc8203e1e00, 0xc8204502b0, 0x7353c0, 0xc8201a0360, 0x32f401, 0x0, 0x0)
	/Users/stacko/Code/go/src/github.com/hashicorp/terraform/helper/schema/resource.go:162 +0x29a
github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc82010e990, 0xc8203e0700, 0xc8203e1e00, 0xc8204502b0, 0x1, 0x0, 0x0)
	/Users/stacko/Code/go/src/github.com/hashicorp/terraform/helper/schema/provider.go:176 +0x1ed
github.com/hashicorp/terraform/terraform.(*EvalApply).Eval(0xc8203e0900, 0xd9e1f0, 0xc820256f70, 0x0, 0x0, 0x0, 0x0)
```

**Comment by stack72** [comment]
The panic is due to the fact that the IP reservation test has no type, yet we expect one in the code

**Comment by ayudemura** [comment]
Putting this on hold.

**Comment by stack72** [comment]
k nps - let me know when you reopen it

**Pull request closed without merging**